### ISSUE TITLE
Next sendout doesn't start automatically. Only after node restart

### ISSUE THREAD

**Issue by kascheri12** [golemfactory/clay#1680]
Just as the title says - while watching the logs, "Next soundout in xxx s" does not show up again until I restart the node.

**Comment by etam** [comment]
This is not a bug.

This message does not need to be visible all the time. If there are some payments to be made, they're done every 10 minutes. This message says that there is a payment waiting to be done.

Example:

```
INFO     [golem.pay                          ] Next sendout in 15 s 
INFO     [golem.pay                          ] Peer count: 25 
INFO     [golem.pay                          ] ETH: 0.0207827 
INFO     [golem.pay                          ] GNT: 445.8333333333333 
INFO     [golem.pay                          ] Next sendout in 2 s 
INFO     [golem.pay                          ] Peer count: 25 
INFO     [golem.pay                          ] ETH: 0.0207827 
INFO     [golem.pay                          ] GNT: 445.8333333333333 
DEBUG:eth.chain.tx      deserialized tx tx=b'8f624b46'
INFO     [golem.pay                          ] Batch payments: bec052, value: 3.333333 
INFO     [golem.pay                          ] Peer count: 25 
INFO     [golem.pay                          ] ETH: 0.0207827 
INFO     [golem.pay                          ] GNT: 442.5 
INFO     [golem.pay                          ] Checking 0xbec0 tx [4] 
INFO     [golem.pay                          ] Peer count: 25 
INFO     [golem.pay                          ] ETH: 0.02004654 
INFO     [golem.pay                          ] GNT: 442.5 
INFO     [golem.pay                          ] Checking 0xbec0 tx [4] 
INFO     [golem.pay                          ] Confirmed 0xbec0: block b7364db00ea85da79cf5e5254be3e7399f4bf96ee57673c44d9b133e4c8d6
06f (1372253), gas 36808, fee 184040000000000                                                                                      
```

After that the message is gone.

Then I created a new task, it got rendered and the message is again appearing, because there is a new payment to be done.

@kascheri12 If you think you found a bug in sending payments, please add more details and reopen this issue.

**Comment by kascheri12** [comment]
I ran a very complete and successful test and have attached the log file here for review. I do believe there is a bug here and it is not sending payments as it should and I will explain below. 

This is how my test went:
* Last night I started a task with 400 subtasks. Woke up this morning and it was in Finished state. No countdown for next sendout was happening at this time.
* Shutdown golemapp
* Start golemapp -- *This is where the log file starts*
  * Note all the payments that are waiting to be processed. 73.3333 GNT reserved at this time for 11 payments. Note 205 more payments waiting to be processed.
* Wait 10 minutes, sendout happens, block confirmed
* Wait approx 30 minutes, nothing is happening.
* Shutdown golemapp
* Start golemapp right away
  * Note 66.6666 GNT reserved at this time for 10 payments. 195 more payments waiting.
* Wait 10 minutes, sendout happens, block confirmed
* Wait 10 minutes, nothing happening
* Line 2006 of log file, I attach golemcli and request 4 new tasks to be computed with 1 subtask each.
  * Line 2282, first subtask is finished. 6.6666 GNT reserved. Sendout starts to count down.
  * Line 2346, 2nd subtask finished. 13.333 GNT reserved.
  * ... 3rd subtask finished. 20.000 GNT reserved.
* Sendout happens for the 20 GNT reserved, block confirmed.
* Waiting, nothing happens
* Line 2867, 4th subtask finishes
  * 6.6666 GNT reserved, Sendout starts to count down again.
* Sendout happens, block confirmed.

At this point, I know the previous payments are not going to happen on their own so I shutdown golemapp again.
* Start golemapp
  * Note 53.3333 GNT reserved for 8 payments. 187 payments still need processing.
* Wait for sendout, it happens, block confirmed.


My conclusion: There are payments that are not being queued to process automatically. In this case I am forcing the backlog to queue up by restarting the node.

All this testing was done on MacOS 10.13.2 with golem-0.9.1 binary.

[20171214_golemapp_log.txt](https://github.com/golemfactory/golem/files/1561101/20171214_golemapp_log.txt)

**Comment by Krigpl** [comment]
@kascheri12 Looks like this is what was happening:
You created a large task that resulted in number of payments.
We tried to add them for sendout one by one.
For each added payment we reserve some amount of ETH for gas.
After a couple of added payments it looks like you're short on ETH for gas so we don't add any more payments (those are these `Low ETH available` messages in the logs).
We make this single sendout and update balances.
Then when you restart the node we read remaining payments from the db and try to add them again for sendout - this time you have enough ETH for a couple of more payments, so we add them, but the cycle happens again.

There are two issues here:
1. Looks like we were overestimating gas cost by quite a lot (from your logs it seems a single transaction used about 0.001 and your balance was 0.01, thus we should be able to send more payments). This logic has been improved recently, so you may want to re-test it.
2. Even if we run out of gas, but later we top up the account we should detect that automatically and add pending payments without the need of restarting the node. This is indeed something we need to implement.

**Comment by Krigpl** [comment]
The linked PR fixes this.
